routes/post.js

```diff
@@ -0,0 +1,115 @@
+const express = require("express")
+const cookieParser = require("cookie-parser")
+const bodyParser = require("body-parser")
+const ObjectId = require('mongoose').Types.ObjectId;
+const { Schema,
+    userSchema,
+    User,
+    sessionSchema,
+    Session,
+    postSchema,
+    Post,
+    replySchema,
+    Reply } = require("../database.js");
+
+const router = express.Router();
+
+router.get("/", async (req, res) => {
+    try {
+        if (!req.user) {
+            res.redirect('/');
+            return;
+        }
+        res.render("newPost", req.user)
+    } catch (err) {
+        res.render('error', { "error-message": err })
+    }
+})
+
+router.get("/:id", async (req, res) => {
+    try {
+        if(!ObjectId.isValid(req.params.id)){
+            res.redirect('/');
+            return;
+        }
+        const currentPost = await Post.findById(req.params.id);
+        if (!currentPost) {
+            res.redirect('/');
+            return;
+        }
+        const maker = await User.findById(currentPost.user.id);
+        const repliesOfThisPostQ = await Reply.find({ post: req.params.id })
+        const repliesOfThisPost = repliesOfThisPostQ.map(item => {
+            return {
+                reply: item.reply,
+                user: { name: item.user.name, id: item.user.id },
+                time: item.created
+            }
+        })
+        // if (!req.user) {
+        //     res.render("post", {
+        //         post: currentPost,
+        //         user: maker,
+        //         replies: repliesOfThisPost
+        //     })
+        //     return;
+        // }
+        res.render("post", {
+            post: currentPost,
+            user: maker,
+            replies: repliesOfThisPost,
+            authenticated: req.user,
+            admin: req.user.admin
```

Comment:
Remember, `req.user` might be undefined here (try accessing this route without a session to see what I mean)! Use `?.` to guard against that, or alternatively `!!req.user && req.user.admin` - with `&&` the second operand will not be invoked if the first operand is false.

```suggestion
            admin: req.user?.admin
```